packages/ember/tests/helpers/helper_registration_test.js

```diff
@@ -1,112 +1,67 @@
+import { moduleFor, ApplicationTestCase } from 'internal-test-helpers';
 import { Controller, Service, inject } from 'ember-runtime';
-import { run } from 'ember-metal';
-import { compile } from 'ember-template-compiler';
-import {
-  Helper,
-  helper,
-  setTemplates,
-  setTemplate
-} from 'ember-glimmer';
-import { Application } from 'ember-application';
-import { Router } from 'ember-routing';
-import { jQuery } from 'ember-views';
-
-let App, appInstance;
-
-QUnit.module('Application Lifecycle - Helper Registration', {
-  teardown() {
-    run(() => {
-      if (App) {
-        App.destroy();
-      }
+import { Helper, helper } from 'ember-glimmer';
 
-      App = appInstance = null;
-      setTemplates({});
-    });
-  }
-});
+moduleFor('Application Lifecycle - Helper Registration', class extends ApplicationTestCase {
+  ['@test Unbound dashed helpers registered on the container can be late-invoked'](assert) {
+    this.addTemplate('application', `<div id='wrapper'>{{x-borf}} {{x-borf 'YES'}}</div>`);
 
-function boot(callback) {
-  run(() => {
-    App = Application.create({
-      name: 'App',
-      rootElement: '#qunit-fixture'
-    });
-
-    App.deferReadiness();
+    let myHelper = helper(params => params[0] || 'BORF');
+    this.add('helper:x-borf', myHelper);
 
-    App.Router = Router.extend({
-      location: 'none'
+    return this.visit('/').then(() => {
+      assert.equal(this.$('#wrapper').text(), 'BORF YES', 'The helper was invoked from the container');
     });
+  }
 
-    // We shouldn't be testing this
-    appInstance = App.__deprecatedInstance__;
-
-    if (callback) { callback(); }
-  });
-
-  let router = appInstance.lookup('router:main');
-
-  run(App, 'advanceReadiness');
-  run(() => router.handleURL('/'));
-}
-
-QUnit.test('Unbound dashed helpers registered on the container can be late-invoked', function() {
-  setTemplate('application', compile('<div id=\'wrapper\'>{{x-borf}} {{x-borf \'YES\'}}</div>'));
-  let myHelper = helper(params => params[0] || 'BORF');
-
-  boot(() => {
-    App.register('helper:x-borf', myHelper);
-  });
-
-  equal(jQuery('#wrapper').text(), 'BORF YES', 'The helper was invoked from the container');
-});
-
-QUnit.test('Bound helpers registered on the container can be late-invoked', function() {
-  setTemplate('application', compile('<div id=\'wrapper\'>{{x-reverse}} {{x-reverse foo}}</div>'));
+  ['@test Bound helpers registered on the container can be late-invoked'](assert) {
```

Comment:
I'm also not sure what `Bound` means for this test. It seems very similar to the previous test.